### DISCUSSION TITLE
API Key 会泄露吗？

### DISCUSSION THREAD

**Discussion by henryhu712** [ChatGPTNextWeb/NextChat#1193] [General]
在 Vercel 上配置了环境变量 API Key，用户使用时，调用 OpenAI 接口会带上这个key啊，这不就泄露了？我的理解是不是有错。

**Comment by Yidadaa** [comment]
服务是在 vercel 服务器上发出的，用户不可能看到对应的请求。

**Comment by jinmiaoluo** [comment]
在 Vercel 服务器上，服务通过用户提供的 OpenAI Access Key（或者在部署服务端代码时通过环境变量注入）与 OpenAI 服务器建立 Server-Sent Events（SSE）数据流连接，为客户端提供相应的 API（代码如下）

https://github.com/Yidadaa/ChatGPT-Next-Web/blob/b23adf9d5dd3b835d245bd471523d438993557db/app/api/chat-stream/route.ts#L50

会对用户的请求进行处理，带上认证信息，转发给 OpenAI 的 API
